File: ide/0.4.6/Phalconry/mvc/plugin/ManagerInterface.zep.php
<?php

namespace Phalconry\Mvc\Plugin;

/**
 * Contract for a plugin manager.
 */
interface ManagerInterface
{

    /**
     * Sets the plugin base namespace.
     *
     * @param string $pluginNamespace 
     * @return void 
     */
    public function setNamespace($pluginNamespace);

    /**
     * Returns the plugin base namespace.
     *
     * @return string 
     */
    public function getNamespace();

    /**
     * Sets the plugin directory path.
     *
     * @throws \InvalidArgumentException if path is invalid.
     * @param string $dirPath 
     * @return void 
     */
    public function setPath($dirPath);

    /**
     * Returns the plugin directory path.
     *
     * @return string|null 
     */
    public function getPath();

    /**
     * Returns the classname of a plugin.
     *
     * @param string $name 
     * @return string 
     */
    public function getPluginClassname($name);

    /**
     * Builds the path to a plugin file.
     *
     * @param string $name 
     * @return string|null 
     */
    public function getPluginFilePath($name);

    /**
     * Registers a plugin as an object or by class.
     *
     * @param string|\Phalconry\Mvc\PluginInterface $classname 
     * @return \Phalconry\Mvc\Plugin\ManagerInterface 
     */
    public function register($classname);

    /**
     * Registers a plugin object.
     *
     * @param \Phalconry\Mvc\PluginInterface $plugin 
     * @return \Phalconry\Mvc\Plugin\ManagerInterface 
     */
    public function add(\Phalconry\Mvc\PluginInterface $plugin);

    /**
     * Returns a plugin by name.
     * If the plugin has not been registered, it will be located.
     *
     * @param string $name 
     * @return \Phalconry\Mvc\PluginInterface|null 
     */
    public function get($name);

    /**
     * Checks if a plugin has been registered.
     *
     * @param string|\Phalconry\Mvc\PluginInterface $plugin 
     * @return boolean 
     */
    public function isRegistered($plugin);

    /**
     * Checks if a plugin has been loaded.
     *
     * @param string|\Phalconry\Mvc\PluginInterface $plugin 
     * @return boolean 
     */
    public function isLoaded($plugin);

    /**
     * Loads a plugin.
     *
     * @param string|\Phalconry\Mvc\PluginInterface $plugin 
     * @return \Phalconry\Mvc\PluginInterface 
     */
    public function load($plugin);

}

File: ext/phalconry/mvc/module/managerinterface.zep.c

#ifdef HAVE_CONFIG_H
#include "../../../ext_config.h"
#endif

#include <php.h>
#include "../../../php_ext.h"
#include "../../../ext.h"

#include <Zend/zend_exceptions.h>

#include "kernel/main.h"


ZEPHIR_INIT_CLASS(Phalconry_Mvc_Module_ManagerInterface) {

	ZEPHIR_REGISTER_INTERFACE(Phalconry\\Mvc\\Module, ManagerInterface, phalconry, mvc_module_managerinterface, phalconry_mvc_module_managerinterface_method_entry);

	return SUCCESS;

}

/**
 * Sets the name of the primary module.
 *
 * @param string moduleName
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, setPrimaryModuleName);

/**
 * Returns the name of the primary module.
 *
 * @return string
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, getPrimaryModuleName);

/**
 * Sets the name of the default module.
 *
 * @param string moduleName
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, setDefaultModuleName);

/**
 * Returns the name of the default module.
 *
 * @return string
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, getDefaultModuleName);

/**
 * Registers an array of module definitions.
 *
 * @param array definitions
 * @return void
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, registerModules);

/**
 * Sets a module in the registry
 *
 * @param \Phalconry\Mvc\ModuleInterface module
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, addModule);

/**
 * Returns a module by name, or the primary module if no name is given
 *
 * @param string name Module name
 * @return \Phalconry\Mvc\ModuleInterface
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, getModule);

/**
 * Checks whether a module exists
 *
 * @param string|\Phalconry\Mvc\ModuleInterface name
 * @return boolean
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, hasModule);

/**
 * Returns the primary module
 *
 * @return \Phalconry\Mvc\ModuleInterface
 *
 * @throws \Phalcon\Mvc\Exception if primary module is not set
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, getPrimaryModule);

/**
 * Loads a module
 *
 * @param \Phalconry\Mvc\ModuleInterface module
 * @return \Phalconry\Mvc\ModuleInterface
 *
 * @throws \Phalcon\Mvc\Exception if module has already been loaded
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, loadModule);

/**
 * Whether a module has been loaded
 *
 * @param string|\Phalconry\Mvc\Module $module
 * @return boolean
 *
 * @throws \InvalidArgumentException if not given a Module object or string
 */
ZEPHIR_DOC_METHOD(Phalconry_Mvc_Module_ManagerInterface, isLoaded);

